packages/hdwallet-provider/src/index.ts

```diff
@@ -131,6 +104,21 @@ class HDWalletProvider {
     const tmpAccounts = this.addresses;
     const tmpWallets = this.wallets;
 
+    // if user supplied the chain id, use that - otherwise fetch it
+    if (typeof chainId !== "undefined" || (chainSettings && typeof chainSettings.chainId !== "undefined")) {
```

Comment:
If the user passed in `null` for the `chainId` or `chainSettings.chainId` we won't fetch the `chainId` from the network, we'll just use `null`. Is that the intent here?

Reply:
Yeah I talked to @gnidan and proposed that at that point it's on them. In the future if people really want to be able to pass in `null` and make a good argument, then I suppose we can allow it. For now I'll claim that this is fine unless someone else wants to argue with me.

Reply:
No, currently it will probably break if they put `null` in there. That is what `chainId` will be set to: that or `undefined`. I was just saying that I don't really want to care about that. Just like if they put some random object as a value for `chainId`, I don't want to care about that either. Unless we want to implement a much more robust input validation utility here.